doc: fix CI codespell warnings

l0rinc opened the pull request:
Can be checked locally by running `test/lint/lint-spelling.py`

l0rinc committed:
doc: fix a few simple codespell warnings

kevkevinpal commented:
Can you update the title and commit to start with `doc:` instead of `CI:`?

since this is just a doc change

jonatack reviewed (commented):
Missing the following one with current codespell 2.3.0

```
src/qt/forms/optionsdialog.ui:344: incomin ==> incoming
```

Perhaps allow "highTS" but not "hights", as the latter could be an misspelling of "heights"

```diff
--- a/test/lint/spelling.ignore-words.txt
+++ b/test/lint/spelling.ignore-words.txt
 hashIn
-hights
+highTS
 inflight
```

l0rinc commented on src/test/fuzz/utxo_snapshot.cpp:
```diff
@@ -72,7 +72,7 @@ void utxo_snapshot_fuzz(FuzzBufferType buffer)
     FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
     auto& setup{*g_setup};
-    bool dirty_chainman{false}; // Re-use the global chainman, but reset it when it is dirty
+    bool dirty_chainman{false}; // Reuse the global chainman, but reset it when it is dirty
```
We have a few cases of both, but I don't recall seeing it spelled this way anywhere else. My vote goes for fixing them instead.

l0rinc replied:
Not sure, in other cases we've just fixed "re-used": https://github.com/bitcoin/bitcoin/pull/28605/files#diff-794891fcb80950b0c81f330c416efdf5d35789fb56a91a18dcc25266d7cdccfdL55

l0rinc replied:
Lol, seems like everyone is talking about this: https://github.com/codespell-project/codespell/issues/3521

l0rinc replied:
Removed the invalid ones and added re-use and incomin

l0rinc replied:
done

l0rinc commented:
> src/qt/forms/optionsdialog.ui:344: incomin ==> incoming

I've seen this one before, it's not a typo and we can't really whitelist it either: `incomin&amp;g` (I think it means that `g` is a shortcut key)

l0rinc commented:
> Probably preferable to avoid needless warnings

Since we seem to ignore many other QT related files and folder, I've added this one to the excludes - what do you think?

fanquake commented:
> The link in the pull description doesn't work, and can be removed.
> Also, you don't need to duplicate the full diff in the pull description.

> Can you update the .. commit to start with doc: instead of CI:?

Did you miss the review feedback here.

fanquake commented:
> Applied the rest of the suggestions.

You missed this one:

> Can you update the .. commit to start with doc: instead of CI:?

l0rinc commented:
> Can you update the .. commit to start with doc: instead of CI:?

done (I interpreted it backwards because of the colons)

fanquake commented on test/lint/lint-spelling.py:
```diff
@@ -14,7 +14,7 @@
 
 IGNORE_WORDS_FILE = 'test/lint/spelling.ignore-words.txt'
-FILES_ARGS = ['git', 'ls-files', '--', ":(exclude)build-aux/m4/", ":(exclude)contrib/seeds/*.txt", ":(exclude)depends/", ":(exclude)doc/release-notes/", ":(exclude)src/qt/locale/", ":(exclude)src/qt/*.qrc", ":(exclude)contrib/guix/patches"]
+FILES_ARGS = ['git', 'ls-files', '--', ":(exclude)build-aux/m4/", ":(exclude)contrib/seeds/*.txt", ":(exclude)depends/", ":(exclude)doc/release-notes/", ":(exclude)src/qt/locale/", ":(exclude)src/qt/forms/optionsdialog.ui", ":(exclude)src/qt/*.qrc", ":(exclude)contrib/guix/patches"]
```
Whitelisting a word seems fine compared to ignoring the spelling for an entire user-facing settings page.

l0rinc replied:
You'd rather we silence "incomin" everywhere? See: https://github.com/bitcoin/bitcoin/pull/30700#issuecomment-2308524353

l0rinc committed:
Update spelling.ignore-words

Removed ba, inflight, keypair and warmup.
Added incomin found in optionsdialog.ui:345 and re-use found in utxo_snapshot.cpp

fanquake reviewed (approved):
ack 837fbca03602982d9f1055ac798711241ea3d0a0 ci output looks fine.

jonatack reviewed (commented):
ACK 837fbca03602982d9f1055ac798711241ea3d0a0
